src/parser/json/dict_parsing_functions.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import datetime
+from typing import Any, Callable, Dict, List
+
+from src.model.typing.constructor_type_errors import ConstructorTypeErrors
+from src.parser.error import SPDXParsingError
+from src.parser.logger import Logger
+
+
+def datetime_from_str(date_str: str) -> datetime:
+    if not isinstance(date_str, str):
+        raise SPDXParsingError([f"Could not convert str to datetime, invalid type: {type(date_str).__name__}"])
+    date = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%SZ")
+    return date
+
+
+def json_str_to_enum_name(json_str: str) -> str:
+    return json_str.replace("-", "_").upper()
+
+
+def construct_or_raise_parsing_error(object_to_construct: Any, args_for_construction: Dict) -> Any:
+    try:
+        constructed_object = object_to_construct(**args_for_construction)
+    except ConstructorTypeErrors as err:
+        raise SPDXParsingError([f"Error while constructing {object_to_construct.__name__}: {err.get_messages()}"])
+    return constructed_object
+
+
+def parse_field_or_log_error(logger: Logger, field: Any, parsing_method: Callable = lambda x: x, default=None,
```

Comment:
I suggest to do `None` handling here also for the case of `optional=False` as I found several places in the code that will throw errors otherwise (I marked a few of them already but it is probably more error-proof to check for `None` at this central position).

Reply:
done